### PR TITLE
Added new filters when adding shipping rates

### PR DESCRIPTION
**Pull request by daigo75**
Ref. https://github.com/woocommerce/woocommerce/issues/20632

### CODE REVIEW

File: includes/abstracts/abstract-wc-shipping-method.php [hunk 1]
```diff
@@ -263,15 +263,16 @@ public function get_rate_id( $suffix = '' ) {
...
 			'id'        => $this->get_rate_id(), // ID for the rate. If not passed, this id:instance default will be used.
 			'label'     => '', // Label for the rate.
 			'cost'      => '0', // Amount or array of costs (per item shipping).
 			'taxes'     => '', // Pass taxes, or leave empty to have it calculated for you, or 'false' to disable calculations.
 			'calc_tax'  => 'per_order', // Calc tax per_order or per_item. Per item needs an array of costs.
 			'meta_data' => array(), // Array of misc meta data to store along with this rate - key value pairs.
 			'package'   => false, // Package array this rate was generated for @since 2.6.0.
-		) );
+			'price_decimals' => wc_get_price_decimals(),
+		), $this ) );
```
Comment (mikejolley) [inline includes/abstracts/abstract-wc-shipping-method.php]: I think $this is in the wrong place?

**Review by mikejolley — CHANGES_REQUESTED** [review]

**Comment by daigo75** [comment]
You're absolutely right. I must have copy/pasted it without noticing, my head is all over the place these days.

I'll upload the fix shortly.

**Comment by daigo75** [comment]
Done. The shipping method (the `$this`) is passed to the filter to provide additional context. Now the variable is in the correct place.

**Comment by mikejolley** [comment]
@daigo75 Can we also use `woocommerce_shipping_method_` prefix on these new hooks?

**Comment by daigo75** [comment]
Of course. Done, as requested.

**Review by mikejolley — APPROVED** [review]

**Pull request merged**